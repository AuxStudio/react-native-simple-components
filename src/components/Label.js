import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

import Icon from '../assets/icons/index';
import styleConstants from '../assets/styleConstants';

import Touchable from './Touchable';

const styles = StyleSheet.create({
    label: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: styleConstants.white,
        borderWidth: 1,
        borderColor: styleConstants.primary,
        borderRadius: 32,
        paddingVertical: 8,
        paddingHorizontal: 16,
        marginBottom: 4,
        marginHorizontal: 4,
        height: 36,
    },
    labelIcon: {
        fontSize: styleConstants.iconFont,
        color: styleConstants.primary,
        marginRight: 8,
    },
    labelText: {
        fontSize: styleConstants.smallFont,
        color: styleConstants.primary,
    },
});

export default (Label = props => {
    /*
        PROPTYPES

        handlePress: PropTypes.func,
        iconName: PropTypes.string,
        labelText: PropTypes.string,
    */

    const label = props.handlePress ? (
        <Touchable onPress={props.handlePress} style={styles.label}>
            <Icon name={props.iconName} style={styles.labelIcon} />
            <Text style={[styles.labelText, styleConstants.primaryFont]}>
                {props.labelText}
            </Text>
        </Touchable>
    ) : (
        <View style={styles.label}>
            <Icon name={props.iconName} style={styles.labelIcon} />
            <Text style={[styles.labelText, styleConstants.primaryFont]}>
                {props.labelText}
            </Text>
        </View>
    );

    return label;
});
