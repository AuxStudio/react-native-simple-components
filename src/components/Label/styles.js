import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: styleConstants.dividerColor,
    paddingVertical: 4,
    paddingHorizontal: 8,
  },
  icon: {
    fontSize: styleConstants.iconFont,
    color: styleConstants.primaryText,
    marginRight: 8,
  },
  text: {
    fontSize: styleConstants.smallFont,
    color: styleConstants.primaryText,
  },
});

export default styles;
