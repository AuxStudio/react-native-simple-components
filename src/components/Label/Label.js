import React from 'react';
import { View, Text } from 'react-native';

import styles from './styles';
import styleConstants from '../assets/styleConstants';

import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
import Touchable from './Touchable';

export default (Label = (props) => {
  /*
        PROPTYPES

        handlePress: PropTypes.func,
        iconName: PropTypes.string,
        customIcon: PropTypes.node,
        text: PropTypes.string,

        showShadow: PropTypes.bool,
        style: PropTypes.node,
        textStyle: PropTypes.node,
        iconStyle: PropTypes.node,
    */

  const shadowStyles = props.showShadow && {
    ...styleConstants.smallShadow,
  };

  const icon = props.customIcon
    ? props.customIcon
    : props.iconName && (
        <MaterialIcon name={props.iconName} style={[styles.icon, props.iconStyle]} />
      );

  const label = props.handlePress ? (
    <Touchable onPress={props.handlePress} style={[styles.container, shadowStyles, props.style]}>
      {icon}
      <Text style={[styles.text, props.textStyle]}>{props.text}</Text>
    </Touchable>
  ) : (
    <View style={[styles.container, shadowStyles, props.style]}>
      {icon}
      <Text style={[styles.text, props.textStyle]}>{props.text}</Text>
    </View>
  );

  return label;
});
