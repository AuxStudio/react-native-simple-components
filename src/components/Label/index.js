import React from 'react';
import PropTypes from 'prop-types';
import { Text, ViewPropTypes } from 'react-native';
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';
import styleConstants from '../../styleConstants';
import Touchable from '../Touchable';

const propTypes = {
  handlePress: PropTypes.func,
  iconName: PropTypes.string,
  customIcon: PropTypes.node,
  text: PropTypes.string,
  showShadow: PropTypes.bool,
  textStyle: Text.propTypes.style,
  iconStyle: Text.propTypes.style,
  style: ViewPropTypes.style,
};

const defaultProps = {
  text: 'Label',
};

const Label = ({
  handlePress,
  iconName,
  customIcon,
  text,
  showShadow,
  textStyle,
  iconStyle,
  style,
}) => {
  const shadowStyles = showShadow && styleConstants.shadows.small;

  const iconComponent = customIcon || (
    <MaterialIcon name={iconName} style={[styles.icon, iconStyle]} />
  );

  return (
    <Touchable
      onPress={handlePress}
      disabled={!handlePress}
      style={[styles.container, shadowStyles, style]}
    >
      {iconComponent}
      <Text style={[styles.text, textStyle]}>{text}</Text>
    </Touchable>
  );
};

Label.propTypes = propTypes;
Label.defaultProps = defaultProps;

export default Label;
