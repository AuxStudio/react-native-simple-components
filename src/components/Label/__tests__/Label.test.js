import React from 'react';
import renderer from 'react-test-renderer';
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';

import Label from '../';

describe('Label', () => {
  it('renders with all props', () => {
    expect(
      renderer.create(
        <Label
          handlePress={jest.fn()}
          iconName="check"
          text="Test"
          showShadow
          iconStyle={{ color: 'red' }}
          textStyle={{ color: 'red' }}
          style={{ backgroundColor: 'blue' }}
        />,
      ),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<Label />)).toMatchSnapshot();
  });

  it('renders with custom icon', () => {
    expect(renderer.create(<Label customIcon={<MaterialIcon name="check" />} />)).toMatchSnapshot();
  });
});
