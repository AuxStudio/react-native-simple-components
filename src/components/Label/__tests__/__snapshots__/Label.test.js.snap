// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders with all props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderColor": null,
      "borderWidth": 0,
      "elevation": 2,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 4,
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.17,
      "shadowRadius": 2,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#212121",
            "fontSize": 24,
            "marginRight": 8,
          },
          Object {
            "color": "red",
          },
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#212121",
          "fontSize": 15,
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`Label renders with custom icon 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E0E0E0",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 4,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#212121",
          "fontSize": 15,
        },
        undefined,
      ]
    }
  >
    Label
  </Text>
</View>
`;

exports[`Label renders with minimum required props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E0E0E0",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
      "paddingVertical": 4,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#212121",
            "fontSize": 24,
            "marginRight": 8,
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#212121",
          "fontSize": 15,
        },
        undefined,
      ]
    }
  >
    Label
  </Text>
</View>
`;
